src/SFML/Graphics/VertexBuffer.cpp

```diff
@@ -0,0 +1,373 @@
+
+    TransientContextLock contextLock;
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, m_buffer));
+
+    // Check if we need to resize or orphan the buffer
+    if (vertexCount >= m_size)
+    {
+        glCheck(GLEXT_glBufferData(GLEXT_GL_ARRAY_BUFFER, sizeof(Vertex) * vertexCount, 0, usageToGlEnum(m_usage)));
+
+        m_size = vertexCount;
+    }
+
+    glCheck(GLEXT_glBufferSubData(GLEXT_GL_ARRAY_BUFFER, sizeof(Vertex) * offset, sizeof(Vertex) * vertexCount, vertices));
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, 0));
+
+    return true;
+}
+
+
+////////////////////////////////////////////////////////////
+bool VertexBuffer::update(const VertexBuffer& vertexBuffer)
+{
+#ifdef SFML_OPENGL_ES
+
+    return false;
+
+#else
+
+    if (!m_buffer || !vertexBuffer.m_buffer)
+        return false;
+
+    TransientContextLock contextLock;
+
+    // Make sure that extensions are initialized
+    sf::priv::ensureExtensionsInit();
+
+    if (GLEXT_copy_buffer)
+    {
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_COPY_READ_BUFFER, vertexBuffer.m_buffer));
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_COPY_WRITE_BUFFER, m_buffer));
+
+        glCheck(GLEXT_glCopyBufferSubData(GLEXT_GL_COPY_READ_BUFFER, GLEXT_GL_COPY_WRITE_BUFFER, 0, 0, sizeof(Vertex) * vertexBuffer.m_size));
+
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_COPY_WRITE_BUFFER, 0));
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_COPY_READ_BUFFER, 0));
+
+        return true;
+    }
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, m_buffer));
+    glCheck(GLEXT_glBufferData(GLEXT_GL_ARRAY_BUFFER, sizeof(Vertex) * vertexBuffer.m_size, 0, usageToGlEnum(m_usage)));
+
+    void* destination = 0;
+    glCheck(destination = GLEXT_glMapBuffer(GLEXT_GL_ARRAY_BUFFER, GLEXT_GL_WRITE_ONLY));
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, vertexBuffer.m_buffer));
+
+    void* source = 0;
+    glCheck(source = GLEXT_glMapBuffer(GLEXT_GL_ARRAY_BUFFER, GLEXT_GL_READ_ONLY));
+
+    std::memcpy(destination, source, sizeof(Vertex) * vertexBuffer.m_size);
+
+    GLboolean sourceResult = GL_FALSE;
+    glCheck(sourceResult = GLEXT_glUnmapBuffer(GLEXT_GL_ARRAY_BUFFER));
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, m_buffer));
+
+    GLboolean destinationResult = GL_FALSE;
+    glCheck(destinationResult = GLEXT_glUnmapBuffer(GLEXT_GL_ARRAY_BUFFER));
+
+    glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, 0));
+
+    if ((sourceResult == GL_FALSE) || (destinationResult == GL_FALSE))
+        return false;
+
+    return true;
+
+#endif // SFML_OPENGL_ES
+}
+
+
+////////////////////////////////////////////////////////////
+void VertexBuffer::clear()
+{
+    if (m_buffer)
+    {
+        TransientContextLock contextLock;
+
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, m_buffer));
+        glCheck(GLEXT_glBufferData(GLEXT_GL_ARRAY_BUFFER, 0, 0, usageToGlEnum(m_usage)));
+        glCheck(GLEXT_glBindBuffer(GLEXT_GL_ARRAY_BUFFER, 0));
+    }
```

Comment:
Can this be a simple call to `create(0)`? That would also avoid the bug of `m_size` not being set to 0 😛 

This would also add proof that this function may not be necessary in the public API.